### ISSUE TITLE
local invoke --env-vars should process .yml files as well - not just .json

### ISSUE THREAD

**Issue by maxfunke** [aws/aws-sam-cli#3113]
### Describe your idea/feature/enhancement

`sam local invoke --env-vars` should also be able to process .yml files.
As of now, it has to be a .json file.

### Proposal

i.e. cloudformation does allow to load environment variables from a .yml file for example with
```yaml
provider:
  environment: ${file(./env.yml)}
```

So it would be handy if I could reuse the file.

Thank you

**Comment by jfuss** [comment]
@maxfunke Thanks for the Feature Request. `--env-vars` does differently than CloudFormation (though your example looks more like Serverless Framework than CloudFormation). For `sam local invoke`, there is a specific format it follows which does not map to CloudFormation at all.

Are you looking for this because you want to inline the `--env-vars` within the Template? If so that would require a new format for `--env-vars` file (which is unlikely we can change without a major version bump). 

Are you just wanting SAM CLI to read an external file, like CloudFormation does? The only way to include a file in CloudFormation is using an [`AWS::Include` Transform](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/create-reusable-transform-function-snippets-and-add-to-your-template-with-aws-include-transform.html), so I assume that is what you are actually looking for here?

**Comment by maxfunke** [comment]
@jfuss you are right I'm actually talking about serverless framework - not cloudformation.
 Yes the later one is what I want to do. Transforming the file so that it matches your demands would be nice.
The main goal would be to reuse the config file, which in my case stores/defines environment variables.

**Comment by qingchm** [comment]
Hey @maxfunke sorry for the late reply we will have this feature request reviewed sometime and let you know if we have it in our queue!
